fpmsyncd/routesync.cpp

```diff
@@ -90,11 +90,15 @@ void RouteSync::onMsg(int nlmsg_type, struct nl_object *obj)
         struct nl_addr *addr = rtnl_route_nh_get_gateway(nexthop);
         unsigned int ifindex = rtnl_route_nh_get_ifindex(nexthop);
 
-        if (addr != NULL)
+        if (addr == NULL)
         {
-            IpAddress nh(*(uint32_t*)nl_addr_get_binary_addr(addr));
-            nexthops += nh.to_string();
+            SWSS_LOG_INFO("Got empty nexthop gateway: %d\n", i);
```

Comment:
Can you check what are the routes? We can then decide how to do it. We should not simply ignore these cases.